services/datamanager/builtin/builtin.go

```diff
@@ -431,8 +437,28 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 		svc.closeCollectors()
 		return err
 	}
-	if cfg.Cloud != nil {
-		svc.partID = cfg.Cloud.ID
+
+	// Check that we have models to download and appropriate credentials.
+	if svcConfig.ModelsToDeploy != nil && cfg.Cloud != nil {
```

Comment:
The previous comment made me also think that we should add a test that directly reads from a new `data_manager_deploy_model.json` to be sure that we're correctly parsing `models_to_deploy` from a config. We inject ModelsToDeploy into the Config format in the current test, but likely need to explicitly add a json parser for Models within `models_to_deploy`

Could you add a TODO in the test for the above? Thank you!

Reply:
done. we now have DATA-493 added above.